sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/MetadataColumn.java

```diff
@@ -0,0 +1,58 @@
+package org.apache.spark.sql.connector.catalog;
```

Comment:
hmm @brkyvz @rdblue don't we require license header for new files?

Reply:
Hmm that's weird RAT misses this.

Reply:
I'll open a new PR to add the license headers.

Reply:
Fixed in https://github.com/apache/spark/pull/30415

Reply:
It was due to incorrect exclusion rule. I made a PR.
- https://github.com/apache/spark/pull/30418